controlplane/kubeadm/internal/machinefilters/machine_filters.go

```diff
@@ -93,10 +105,32 @@ func OwnedControlPlaneMachines(controlPlaneName string) Func {
 		if controllerRef == nil {
 			return false
 		}
-		return controllerRef.Kind == "KubeadmControlPlane" && controllerRef.Name == controlPlaneName
+		return controllerRef.Kind == "KubeadmControlPlane" && controllerRef.Name == owner.GetName()
 	}
 }
 
+// ControlPlaneMachines returns a filter to find all control plane machines for a cluster, regardless of ownership.
+// Usage: managementCluster.GetMachinesForCluster(ctx, cluster, machinefilters.ControlPlaneMachines(cluster.Name))
+func ControlPlaneMachines(clusterName string) func(machine *clusterv1.Machine) bool {
+	selector := ControlPlaneSelectorForCluster(clusterName)
+	return func(machine *clusterv1.Machine) bool {
+		if machine == nil {
+			return false
+		}
+		return selector.Matches(labels.Set(machine.Labels))
+	}
+}
+
+// AdoptableControlPlaneMachines returns a filter to find all un-controlled control plane machines.
+// Usage: managementCluster.GetMachinesForCluster(ctx, cluster, AdoptableControlPlaneMachines(cluster.Name, controlPlane))
+func AdoptableControlPlaneMachines(clusterName string) func(machine *clusterv1.Machine) bool {
+	return And(
+		Not(IsNil),
+		ControlPlaneMachines(clusterName),
+		Not(HasControllerRef),
+	)
```

Comment:
That's being checked before adoption in the controller, along with the version.

I figured it would be better in those cases to have the KCP controller emit an "I don't know what to do" error in a loop than proceed as if the existing CP machines weren't there.

Reply:
I was thinking this could impact the reconciliation of a KCP instance if there is some user mistake. Let's say for example, some user (or controller) creates a machine with a control-plane label, KCP controller will be stuck in trying to adopt the machines

Reply:
Yeah, that's what I think I would expect as a user: if I make up a control plane node, I would expect the KCP to either take it over or tell me why it can't.

The alternative I saw a few times in testing was that it would happily spin up a second control plane, with the same certs, behind the same LB, but backed by a different etcd.